lib/ruby_lsp/server.rb

```diff
@@ -320,11 +320,11 @@ def text_document_did_open(message)
           language_id: language_id,
         )
 
-        if document.past_expensive_limit?
+        if document.past_expensive_limit? && text_document[:uri].scheme == "file"
           send_message(
             Notification.new(
-              method: "window/showMessage",
-              params: Interface::ShowMessageParams.new(
+              method: "window/logMessage",
+              params: Interface::LogMessageParams.new(
                 type: Constant::MessageType::WARNING,
                 message: "This file is too long. For performance reasons, semantic highlighting and " \
```

Comment:
Since this is now 'separated' from the active editor tab, I think we should include the path in the message.